File: Leetcode/hashmap/771_JewelAndStone/Solution.java
class Solution {
    public int numJewelsInStones(String J, String S) {
        Set<Character> set = new HashSet<>();
        for(Character c: J.toCharArray()){
            set.add(c);
        }
        int res = 0;
        for(Character c: S.toCharArray()){
            if(set.contains(c))
                res++;
        }
        return res;
    }
}

File: Leetcode/hashmap/771_JewelAndStone/Solution.py
class Solution(object):
    def numJewelsInStones(self, J, S):
        """
        :type J: str
        :type S: str
        :rtype: int
        """
        """
        s = set()
        for c in J:
            s.add(c)
        res = 0
        for c in S:
            if c in s:
                res+=1
        return res
        """
        return sum(map(S.count, J))
